From 94df6e440202de93d44ec3b154695a17ec74b7b2 Mon Sep 17 00:00:00 2001
From: Thomas Adam <thomas@xteddy.org>
Date: Sat, 10 Dec 2016 12:11:59 +0000
Subject: [PATCH] query: aggregate license information

Currently, multiple packages queried with:

    query '%n %L'

will output packages on multiple lines, skewing counts with "wc -l".  Instead,
aggregate all of the licenses of a package and output that.

Additionally, handle printing packages with no license information by using an
empty string.

Fixes #1510
---
 libpkg/pkgdb_iterator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libpkg/pkgdb_iterator.c b/libpkg/pkgdb_iterator.c
index addbc09..df73c6e 100644
--- libpkg/pkgdb_iterator.c
+++ libpkg/pkgdb_iterator.c
@@ -507,7 +507,7 @@ static int
 pkgdb_load_license(sqlite3 *sqlite, struct pkg *pkg)
 {
 	const char	 sql[] = ""
-		"SELECT name"
+		"SELECT ifnull(group_concat(name, ', '), '') AS name"
 		"  FROM pkg_licenses, licenses AS l"
 		"  WHERE package_id = ?1"
 		"    AND license_id = l.id"
